flow/core/params.py

```diff
@@ -270,6 +278,19 @@ def add(self,
         lane_change_params : flow.core.params.SumoLaneChangeParams
             Params object specifying attributes for Sumo lane changing model.
         """
+
+        # determine which energy model to use
+        if "prius" in veh_id:
+            energy_model = env.k.PriusEnergy.get_instantaneous_power()
+        elif "tacoma" in veh_id:
+            energy_model = env.k.TacomaEnergy.get_instantaneous_power()
+        elif "ev" in veh_id:
+            energy_model = env.k.PDMElectric.get_instantaneous_power()
+        else:
+            energy_model = env.k.PDMCombustionEngine.get_instantaneous_power()
+   
+    
+
```

Comment:
This is where assignment of energy models happen inside the VehicleParams().  I set PDMCombustionEngine model as the default energy model.

Reply:
rather than instantiating the instantaneous power method here, you should just provide the class names, e.g.
```
if "prius" in veh_id:
    energy_class_name = PriusEnergy
```

when the actual vehicles are added to the simulation domain (i think in `add_departed()`?), there you will instantiate the class. i'm not sure about the design choice, but you might have a dictionary of energy classes (say `energy_classes = dict()`), and every time a new vehicle is added, you say something like:
```
energy_classes[veh_id] = energy_class_name(kernel, **kwargs)
```
and this is where you need to figure out the args/kwargs stuff, to pass `initial_soc` if needed.